sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -253,6 +254,15 @@ class QueryExecution(
 
     // trigger to compute stats for logical plans
     try {
+      optimizedPlan.transform {
```

Comment:
shall we use `foreach` instead of `transform`?

Reply:
> shall we use `foreach` instead of `transform`?

Yea, seems foreach more better in here